framework/include/cppmicroservices/detail/BundleResourceBuffer.h

```diff
@@ -66,6 +66,15 @@ namespace cppmicroservices
 
           private:
             std::unique_ptr<BundleResourceBufferPrivate> d;
+
+            friend class BundleResourceBufferTest;
+            friend class BundleResourceBufferTest_ConstructionWithEmptyData_Test;
+            friend class BundleResourceBufferTest_UnderflowAndUflow_Test;
+            friend class BundleResourceBufferTest_Pbackfail_Test;
+            friend class BundleResourceBufferTest_Showmanyc_Test;
+            friend class BundleResourceBufferTest_SeekoffAndSeekpos_Test;
+            friend class BundleResourceBufferTest_NewlineConversion_Test;
+            friend class BundleResourceBufferTest_RemoveLastNewline_Test;
```

Comment:
I would like to avoid lots of `friend class` declarations when possible. From a design perspective, the shipping code shouldn't be modified for the sole purpose of accommodating tests. In this case the friend classes could be made unnecessary if the tests used the public APIs of the `std::streambuf` class to test the private methods.

Reply:
Removed and replaced with a single `friend class BundleResourceBufferTestSuite`.  Would it be better if I removed that as well?